src/main/java/uk/gov/pay/ledger/util/pagination/PaginationBuilder.java

```diff
@@ -122,6 +124,7 @@ private void buildLinks(long lastPage) {
     }
 
     private String uriWithParams(String queryParams) {
+        queryParams = URLEncoder.encode(queryParams, StandardCharsets.UTF_8).replace("%3D", "=").replace("%26","&");
```

Comment:
This is a bit 😬

It looks like the `queryParams` string comes from the earlier `searchParams.buildQueryParamString(…)` call, where `searchParams` is a subclass of `SearchParams`, which is a rich object that knows the keys and values for each parameter.

So could we instead modify this `buildQueryParamString(…)` method to encode the values as it appends them to string (which, honestly, it feels like it should have been doing already and I’m not really sure how we survived this long without it)?

Reply:
That makes a lot more sense. I'd had a look at `buildQueryParamString(…)` but wasn't entirely sure of its function (just returns a Long ?).